lib/parsers.js

```diff
@@ -102,6 +99,25 @@ function getContextFromRequest (req, conf, type) {
   return context
 }
 
+/**
+ * Extract appropriate `{transaction,error}.context.response` from an HTTP
+ * response object. This handles header redaction according to the agent config.
+ *
+ * @param {Object} res - Typically `res` is a Node.js `http.OutgoingMessage`
+ *    (https://nodejs.org/api/http.html#class-httpoutgoingmessage).
+ *    However, some cases (e.g. Lambda and Azure Functions instrumentation)
+ *    create a pseudo-res object that matches well enough for this function.
+ *    Some relevant fields: (TODO: document all used fields)
+ *    - `headers` - Required. An object.
```

Comment:
*Is* `res.headers` now required? Note that there is a code path in lambda.js that sets a pseudo `trans.res` that does *not* have a `headers` field: https://github.com/elastic/apm-agent-nodejs/blob/9f13472d69523109d69315c6bb212957e46809cb/lib/lambda.js#L225-L228

What happens in that case?

Reply:
I added commit d4c5aae2007b2586cba5a90b322eaf7cd6638266 that shows the headers aren't *required*, which sanity checks that this won't crash in that case.